packages/buttons/src/views/icon-button/ChevronButton.js

```diff
@@ -46,7 +46,7 @@ ChevronButton.propTypes = {
   hovered: PropTypes.bool,
   active: PropTypes.bool,
   /** Callback for reference of the native button element */
-  buttonRef: PropTypes.func,
+  buttonRef: PropTypes.any,
```

Comment:
Is `any` really the appropriate choice here? It should only be an object or a function, right?

```
In the other hand, if you actually want to enforce the type of an actual ref, for example a prop that is forwarded to another component's ref, you have to allow both on object shaped by React.createRef() (The preferred way to create ref in React components, see type in react code here) and function types.

Object shape: PropTypes.shape({ current: PropTypes.instanceOf(Element) }) is for refs created using React.createRef()
PropTypes.func is for callback refs as used in the above question
Which gives the following PropType definition:

forwardRef: PropTypes.oneOfType([
  PropTypes.func, 
  PropTypes.shape({ current: PropTypes.instanceOf(Element) })
])
```

Reply:
We have avoided the `shape` prop-type in other packages which is why I went with `any` here.

We ended up removing them as they actually don't do anything in the `v6+` world.